server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -150,4 +188,44 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
 
     }
+
+    private void validate(
+        ClusterState currentState,
+        Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap,
+        boolean restoreAllShards
+    ) {
+        String errorMsg = "cannot restore index [%s] because an open index with same name already exists in the cluster.";
```

Comment:
The condition checked below is for same UUID also. Should we include the same in the error message?

Reply:
Yeah, thanks for catching this. will update

Reply:
Also, you need to provide the resolution i.e. either delete or close the index to restore it from remote.

Reply:
we wont be able to restore closed indices during full cluster restore as the index metadata is already present. to restore a closed index we need to trigger this api without clusterUUID. 

if we give the suggestion to user to delete the index then this would publish a new cluster state to remote and if the cluster UUID is same for current and previous cluster state from which we need to restore, the latest remote state would change

Reply:
We shouldn't allow cluster state restore in manual mode if the UUID is same as local UUID. Any time QuorumLoss is recovered, a new clusterUUID would be generated.

Reply:
@soosinha I've updated the error message to indicate uuid match as well.

@shwetathareja i've added a check for clusterUUID being same to the local cluster UUID